assets/app/views/directives/osc-key-values.html

```diff
@@ -2,18 +2,27 @@
     <div class="form-inline labels-edit" ng-show="editable">
       <form class="edit-label" name="form" novalidate>
         <div class='form-group' ng-class="{'has-error': form.key.$error.oscKeyValid}">
-          <input class="form-control" type="text" name="key" placeholder="{{keyTitle}}" ng-model="key" 
+          <input class="form-control"
+                 type="text"
+                 name="key"
+                 placeholder="{{keyTitle}}"
+                 ng-model="key"
+                 ng-keyup="toggleAdd()"
                  osc-input-validator="key">
         </div>
         <div class="form-group" ng-class="{'has-error': form.value.$error.oscValueValid}">
```

Comment:
@benjaminapetersen you can write a custom one yourself: http://www.benlesh.com/2012/12/angular-js-custom-validation-via.html

or we can think long term and consider pulling in either:
angular-ui's ui-validator package http://htmlpreview.github.io/?https://github.com/angular-ui/ui-validate/master/demo/index.html
or
angular-validator https://github.com/turinggroup/angular-validator